core/extensions.rs

```diff
@@ -257,120 +273,107 @@ macro_rules! extension {
     }
 
     impl $name {
-      #[inline(always)]
-      fn ext() -> $crate::ExtensionBuilder {
-        $crate::Extension::builder_with_deps(stringify!($name), &[ $( $( stringify!($dep) ),* )? ])
-      }
-
-      /// If ESM or JS was specified, add those files to the extension.
-      #[inline(always)]
-      #[allow(unused_variables)]
-      fn with_js(ext: &mut $crate::ExtensionBuilder) {
-        $( ext.esm(
-          $crate::include_js_files!( $name $( dir $dir_esm , )? $( $esm , )* )
-        ); )?
-        $( ext.esm(
-          $crate::include_js_files_with_specifiers!( $name $( dir $dir_esm2 , )? $( ( $esm_specifier, $esm_file) , )* )
-        ); )?
-        $(
-          ext.esm_entry_point($esm_entry_point);
-        )?
-        $( ext.js(
-          $crate::include_js_files!( $name $( dir $dir_js , )? $( $js , )* )
-        ); )?
+      fn ext $( <  $( $param : $type + 'static ),+ > )?() -> $crate::Extension {
+        #[allow(unused_imports)]
+        use $crate::Op;
+        $crate::Extension {
+          // Computed at compile-time, may be modified at runtime with `Cow`:
+          name: stringify!($name),
+          deps: &[ $( $( stringify!($dep) ),* )? ],
+          js_files: std::borrow::Cow::Borrowed(&$crate::or!($($crate::include_js_files!( $name $( dir $dir_js , )? $( $js , )* ))?, [])),
+          esm_files: std::borrow::Cow::Borrowed(&$crate::or!($($crate::include_js_files!( $name $( dir $dir_esm , )? $( $esm $( with_specifier $esm_specifier )? , )* ))?, [])),
+          esm_entry_point: $crate::or!($(Some($esm_entry_point))?, None),
+          ops: std::borrow::Cow::Borrowed(&[$($(
+            $( #[ $m ] )*
+            $( $op )::+ $( :: < $($op_param),* > )? :: DECL
+          ),+)?]),
+          external_references: std::borrow::Cow::Borrowed(&[ $( $external_reference ),* ]),
+          // Computed at runtime:
+          opstate_fn: None,
+          middleware_fn: None,
```

Comment:
Looking at this more closely, these should probably also be converted to `fn()`. What do you think?